cogs/device.py

```diff
@@ -134,6 +134,9 @@ async def add_device(self, ctx: commands.Context) -> None:
                         device['ecid'] = answer[2:]
                     else:
                         device['ecid'] = answer
+                    
+                    if str(device['ecid'].isdecimal()):
```

Comment:
```suggestion
                    if str(device['ecid']).isdecimal():
```
This is going to match every time because you’re casting the boolean result of isdecimal to a string.